File: LeetCode/Difficulty/Medium/java/199.binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private List<Integer> result = new ArrayList<>();

    void solve(TreeNode root, int level) {
        if (root == null) return;

        if (result.size()-1 < level)
            result.add(root.val);
        else
            result.set(level, root.val);

        solve(root.left, level+1);
        solve(root.right, level+1);
    }

    public List<Integer> rightSideView(TreeNode root) {
        solve(root, 0);
        return result;
    }
}

File: LeetCode/Difficulty/Medium/js/199.binary-tree-right-side-view.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var rightSideView = function(root) {
    if(!root) return [];
    const stack = [];
    const result = [];
    let printLevel = 0;
    stack.push({node:root, level:0});
    
    while(stack.length!=0) {
        const {node, level} = stack.pop();
        if(level === printLevel){
            result.push(node.val);
            printLevel++;
        }
        
        if(node.left != null){
            stack.push({node: node.left, level: level+1});
        }
        if(node.right != null){
            stack.push({node: node.right, level: level+1});
        }

    }
    return result;
};
